pkg/handlers/session.go

```diff
@@ -75,12 +76,23 @@ func requireValidSession(kotsStore store.Store, w http.ResponseWriter, r *http.R
 	}
 
 	if time.Now().After(sess.ExpiresAt) {
+		if err := kotsStore.DeleteSession(sess.ID); err != nil {
+			logger.Error(errors.Wrapf(err, "session expired. failed to delete expired session %s", sess.ID))
+		}
 		err := errors.New("session expired")
 		response := ErrorResponse{Error: err.Error()}
 		JSON(w, http.StatusUnauthorized, response)
 		return nil, err
 	}
 
+	// extend the session by an hour, if user is still active after an hour
+	if delta := time.Until(sess.ExpiresAt); delta.Hours() <= (SessionTimeout - 1*time.Hour).Hours() {
+		sess.ExpiresAt = sess.ExpiresAt.Add(time.Hour * 1)
+		if err := kotsStore.UpdateSessionExpiresAt(sess.ID, sess.ExpiresAt); err != nil {
+			logger.Error(errors.Wrapf(err, "failed to update session expiry %s", sess.ID))
```

Comment:
logging the error instead of returning it.
Since it would cause user inconvenience as their session is still valid and active, this may cause a false positive.